hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/protocolPB/OzoneManagerRequestHandler.java

```diff
@@ -378,6 +387,30 @@ private InfoBucketResponse infoBucket(InfoBucketRequest request)
     return resp.build();
   }
 
+  @RequestFeatureValidator(
+      conditions = ValidationCondition.OLDER_CLIENT_REQUESTS,
+      processingPhase = RequestProcessingPhase.POST_PROCESS,
+      requestType = Type.InfoBucket
+  )
+  public static OMResponse disallowInfoBucketResponseWithECReplicationConfig(
```

Comment:
Right, this looks like a new optional field added to the BucketInfo proto. If the old client gets it I would think protobuf would ignore the field and not throw an error.

Reply:
It is an interesting question... I haven't checked deeply what would happen...
But... I firmly believe two things can happen:
1. proto libs initialize lazily and unless someone does not reach the ECReplicationConfig it does not try to initialize it
2. proto libs read through and parse the whole message upon arrival

In the first case, we would still need to ensure that the client will never read the value, which is tedious, and have to be checked in future commits as well up until it is released, so this would be something to check in the release in order to ensure the client does not fail. (This does not seem to be failsafe, and also it is not the easiest task either to write tests for all code branches or to go over it by hand and understand the code.)
So in the first case the client might fail if there is an ECReplicationConfig somehow in the result,
While in the second case the client will fail inevitably when there is an ECReplicaitonConfig in the result.

Based on this I think it is just safer and easier to ensure that we do not serve an ECReplicationConfig from the server side.

Reply:
My understanding is that even if the OM sets DefaultReplicationConfig the old client will still parse the proto without error and ignore the extra field. I think it would be good to confirm the behavior of an old client doing list/info on an EC bucket and it looks like that is listed as a work item [here](https://github.com/apache/ozone/pull/3262#issuecomment-1088361663), so let's wait and see what the results are.

Reply:
Yes you were right test showed it and now bucket info and listing is enabled